### DISCUSSION TITLE
Backing up and restoring rules from the terminal.

### DISCUSSION THREAD

**Discussion by HelloGrayson** [evilsocket/opensnitch#1083] [General]
I'd like to automate the backup and restore of my Opensnitch rules from the terminal.

Do rules get written and read from a certain path in a db/config file? I see that the GUI makes this functionality available under the hamburger menu, so I'm hoping the same is possible from the shell.

**Comment by gustavo-iniguez-goya** [comment]
Hi @HelloGrayson ,

Sorry, probably you already figured it out. Anyway, the rules are saved by the default to `/etc/opensnitchd/rules/`. The GUI stores the rules in the db, but only as a "local"/backup copy of what the daemon has.

You can copy the rules from that directory, and restore them when needed. The daemon will reload them automatically once copied.

**Comment by HelloGrayson** [comment]
I'd like to request that Opensnitch allow for setting the configuration path to $HOME so that users can backup and restore without sudo access or the need to modify the root filing system. 

The Linux desktop ethos seems to be heading towards immutable images (ala silverblue, vanilla, etc) which encourages users to place all of their customization within $HOME, leaving the root filesystem in a default state. I have a full backup/restore program [here](https://github.com/HelloGrayson/workstation) using Chezmoi+restic; this dataset is now the red-headed step child.

Thanks for your consideration.
